internal/controller/component_build_controller.go

```diff
@@ -184,6 +189,62 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		if err := r.EnsureBuildPipelineServiceAccount(ctx, &component); err != nil {
 			return ctrl.Result{}, err
 		}
+
+		// ensure namespace secret forced by annotation, useful for migration
```

Comment:
Here and other places: for people without deep knowledge of the architecture, it's not clear what namespace secret and SA are. Let's use more clear names (and in comments too).

Reply:
ok all updated and unfied